draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -959,37 +978,41 @@ For a given request ("request"), the following algorithm returns `same-site` or
 2.  Let `site` be `request`'s client's "site for cookies" (as defined in the
     following sections).
 
-3.  Let `target` be the registrable domain of `request`'s current url.
+3.  Let `target` be the origin of `request`'s current url.
 
-4.  If `site` is an exact match for `target`, return `same-site`.
+4.  If `site` is same-site with `target`, return `same-site`.
 
 5.  Return `cross-site`.
 
-The request's client's "site for cookies" is calculated depending upon its
-client's type, as described in the following subsections:
+Note: The request's URL when establishing a WebSockets connection {{RFC6455}}
+has scheme "http" or "https", rather than "ws" or "wss". See {{FETCH}} which
+maps schemes when constructing the request. This allows same-site cookies to be
+sent with WebSockets.
+
+The request's client's "site for cookies" is an origin. It is calculated
+depending upon its client's type, as described in the following subsections:
 
 ### Document-based requests {#document-requests}
 
 The URI displayed in a user agent's address bar is the only security context
 directly exposed to users, and therefore the only signal users can reasonably
-rely upon to determine whether or not they trust a particular website. The
-registrable domain of that URI's origin represents the context in which a user
-most likely believes themselves to be interacting. We'll label this domain the
-"top-level site".
+rely upon to determine whether or not they trust a particular website. Thus we
+base the notion of "site for cookies" on the top-level URL. We'll label this
+URI's origin the "top-level origin".
 
 For a document displayed in a top-level browsing context, we can stop here: the
-document's "site for cookies" is the top-level site.
+document's "site for cookies" is the top-level origin.
 
 For documents which are displayed in nested browsing contexts, we need to audit
 the origins of each of a document's ancestor browsing contexts' active documents
 in order to account for the "multiple-nested scenarios" described in Section 4
-of {{RFC7034}}. A document's "site for cookies" is the top-level site if and
-only if the document and each of its ancestor documents' origins have the same
-registrable domain as the top-level site. Otherwise its "site for cookies" is
-the empty string.
+of {{RFC7034}}. A document's "site for cookies" is the top-level origin if and
+only if the document and each of its ancestor documents' origins are same-site
+with the top-level origin. Otherwise its "site for cookies" is a freshly
+generated globally unique identifier.
```

Comment:
Perhaps add a reference linking "freshly generated globally unique identifier" to opaque origins?

Reply:
Defined this further up the doc.